sdk/python/feast/infra/registry/remote.py

```diff
@@ -39,8 +41,16 @@ def __init__(
         registry_config: Union[RegistryConfig, RemoteRegistryConfig],
         project: str,
         repo_path: Optional[Path],
+        auth_config: AuthConfig,
     ):
-        self.channel = grpc.insecure_channel(registry_config.path)
+        if auth_config.type in ["oidc", "kubernetes"]:
```

Comment:
Do we need to do the same for oidc auth type?

Reply:
maybe we use the constant for this  like `if auth_config.type in [OIDC, KUBERNETES]` as it has usage other place too. We can create another PR for this change if it required more time.